controllers/authenticator.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/alexedwards/scs"
+	"go.uber.org/zap"
+)
+
+type Authenticator struct {
+	K8sClient      AuthenticatingClient
+	SessionManager *scs.Manager
+}
+
+func (a Authenticator) tokenReview(token string, req *http.Request) (bool, error) {
+	//TODO not working. temporary disabled.
+	//review := v1.TokenReview{
+	//	Spec: v1.TokenReviewSpec{
+	//		Token: token,
+	//	},
+	//}
+	//
+	//ctx := WithAuthIntoContext(token, req.Context())
+	//
+	//if err := a.K8sClient.Create(ctx, &review); err != nil {
+	//	zap.L().Error("token review error", zap.Error(err))
+	//	return false, err
+	//}
+	//
+	//zap.L().Debug("token review result", zap.Stringer("review", &review))
+	//return review.Status.Authenticated, nil
+	return true, nil
```

Comment:
does it mean that it allow any token?

Reply:
Yes. I've explained it in the description.